src/pokemon.c

```diff
@@ -2538,15 +2552,7 @@ u32 GetBoxMonData3(struct BoxPokemon *boxMon, s32 field, u8 *data)
             break;
         case MON_DATA_TERA_TYPE:
         {
-            if (substruct0->teraType == 0)
```

Comment:
Why remove this section? Any types that aren't set by previous saves should default to something, hence the personality.

Reply:
Oh, I completely forgot about that case. That's kind of annoying

Reply:
I feel like we should do this somewhere else though

Reply:
I moved it to GetTeraType since its pretty much the same